### PR TITLE
make nonceTransactionsHandler with v1 & v2 for backwards compatibility

### PR DESCRIPTION
**Pull request by dragos-rebegea**

### CODE REVIEW

File: interactors/interface.go [hunk 1]
```diff
@@ -29,3 +29,20 @@ type AddressNonceHandler interface {
 	SendTransaction(ctx context.Context, tx *data.Transaction) (string, error)
 	DropTransactions()
 }
+
+// TransactionNonceHandlerV1 is able to handle the
```
Comment (iulianpascalau) [inline interactors/interface.go:33]: unfinished comments here and on L42

File: interactors/nonceHandlerV1/addressNonceHandler.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
+package nonceHandlerV1
```
Comment (iulianpascalau) [inline interactors/nonceHandlerV1/addressNonceHandler.go:1]: compared, looks good :+1:

File: interactors/nonceHandlerV1/nonceTransactionsHandler.go [hunk 3]
```diff
@@ -0,0 +1,171 @@
+package nonceHandlerV1
```
Comment (iulianpascalau) [inline interactors/nonceHandlerV1/nonceTransactionsHandler.go:1]: compared, looks good :+1:

**Review by iulianpascalau — COMMENTED** [review]
looks good :+1:

File: interactors/nonceHandlerV2/nonceTransactionsHandler.go [hunk 4]
```diff
@@ -56,17 +60,20 @@ func NewNonceTransactionHandler(proxy Proxy, intervalToResend time.Duration, che
 }
 
 // ApplyNonce will apply the nonce to the given ArgCreateTransaction
-func (nth *nonceTransactionsHandler) ApplyNonce(ctx context.Context, address core.AddressHandler, txArgs *data.ArgCreateTransaction) error {
+func (nth *nonceTransactionsHandlerV2) ApplyNonce(ctx context.Context, address core.AddressHandler, txArgs *data.ArgCreateTransaction) error {
+	if check.IfNil(address) {
```
[the remark this answers is not in the document]
Comment (dragos-rebegea) [reply interactors/nonceHandlerV2/nonceTransactionsHandler.go:64]: yes, indeed. The txArgs has to be check the second time :+1:

**Comment by dragos-rebegea** [comment]
diffs between current implementation and main implementation:
nonceTransactionsHandler.go

[interactors/addressNonceHandler.go](https://raw.githubusercontent.com/ElrondNetwork/elrond-sdk-erdgo/main/interactors/addressNonceHandler.go) vs [nonceHandlerV1/addressNonceHandler.go](https://raw.githubusercontent.com/ElrondNetwork/elrond-sdk-erdgo/nonceTransactionsHandler-v1-v2/interactors/nonceHandlerV1/addressNonceHandler.go): https://www.diffchecker.com/W3Y8TZ8v
[interactors/nonceTransactionsHandler](https://raw.githubusercontent.com/ElrondNetwork/elrond-sdk-erdgo/main/interactors/nonceTransactionsHandler.go) vs [nonceHandlerV1/nonceTransactionsHandler](https://raw.githubusercontent.com/ElrondNetwork/elrond-sdk-erdgo/nonceTransactionsHandler-v1-v2/interactors/nonceHandlerV1/nonceTransactionsHandler.go): https://www.diffchecker.com/rBSS1msq

File: interactors/interface.go (hunk 1 above)
Comment (dragos-rebegea) [reply interactors/interface.go]: done

File: aggregator/notifees/elrondNotifee_test.go [hunk 5]
```diff
@@ -27,7 +27,7 @@ func createMockArgsElrondNotifee() ArgsElrondNotifee {
 	return ArgsElrondNotifee{
 		Proxy:           &testsCommon.ProxyStub{},
 		TxBuilder:       &testsCommon.TxBuilderStub{},
-		TxNonceHandler:  &testsCommon.TxNonceHandlerStub{},
+		TxNonceHandler:  &testsCommon.TxNonceHandlerV2Stub{},
```
[the remark this answers is not in the document]
Comment (dragos-rebegea) [reply aggregator/notifees/elrondNotifee_test.go:30]: ElrondNotifee for aggregator will use v2

**Review by sstanculeanu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**